testtools/files.go

```diff
@@ -188,191 +209,189 @@ var (
 //            └── sourceFile.foo
 //            └── BUILD.in --> BUILD file prior to running gazelle.
 //            └── BUILD.out --> BUILD file expected after running gazelle.
-func TestGazelleGenerationOnPath(t *testing.T, args *TestGazelleGenerationArgs) {
-	t.Run(args.Name, func(t *testing.T) {
-		t.Helper() // Make the stack trace a little bit more clear.
-		if args.BuildInSuffix == "" {
-			args.BuildInSuffix = ".in"
-		}
-		if args.BuildOutSuffix == "" {
-			args.BuildOutSuffix = ".out"
+func TestGazelleGenerationOnPath(t TB, args *TestGazelleGenerationArgs) {
+	t.Helper() // Make the stack trace a little bit more clear.
```

Comment:
It looks like quite a bit more stuff in this PR than changing the test to take a `TB` interface. If you'd like to change the inner workings of this file I would like to do it in a different PR than the one where we expand the accepted interface here.

Reply:
I reverted the changes, but then I am not able to have a common interface for `testing.T` (`Run(name string, f func(t *T)) bool`) and `testing.B` (`Run(name string, f func(b *B)) bool`).

So I kept the `TestGazelleGenerationOnPath` function as it was (`func(t *testing.T, args *TestGazelleGenerationArgs)`)

I could imagine something using generics but it brings some complexity that we maybe do not want?